app/views/layouts/application.html.erb

```diff
@@ -5,12 +5,33 @@
     <meta name="viewport" content="width=device-width,initial-scale=1">
     <%= csrf_meta_tags %>
     <%= csp_meta_tag %>
+    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
 
     <%= stylesheet_link_tag 'application', media: 'all', 'data-turbolinks-track': 'reload' %>
+    <%= stylesheet_pack_tag 'application', 'data-turbolinks-track': 'reload' %>
     <%= javascript_pack_tag 'application', 'data-turbolinks-track': 'reload' %>
   </head>
 
   <body>
+    <% if notice %>
+      <p class="alert alert-success"><%= notice %></p>
+    <% end %>
+
+    <% if alert %>
+      <p class="alert alert-danger"><%= alert %></p>
+    <% end %>
+
+     <% if user_signed_in? %>
+      
+      Logged in as: <strong><%= current_user.email %> </strong>
+      <%= link_to 'Edit Profile', edit_user_registration_path,
+                                                 class: 'navbar-link btn btn-warning' %>
+      <%= link_to 'Log Out', destroy_user_session_path, method: :delete,
+                                                 class: 'navbar-link btn btn-danger' %>
+    <% else %>
+      <%= link_to 'Sign Up', new_user_registration_path, class: 'navbar-link btn btn-primary' %>
+      <%= link_to 'Log In', new_user_session_path, class: 'navbar-link btn btn-success' %>
+    <% end %>
     <%= yield %>
   </body>
 </html>
```

Comment:
remove extra lines, use single quotes, make partials of flash messages and nav_bar